### DISCUSSION TITLE
can this work for c++

### DISCUSSION THREAD

**Discussion by ryzendew** [AvaloniaUI/Avalonia#15007] [General]
suyu devs are wanting to rebuild the GUI and i want them to use ava but is there any way to use it on a c++ app?

**Comment by thevortexcloud** [comment]
Technically, yes. Should you? Probably not. You need to build Avalonia with AOT so it becomes a native library instead of a .NET one. There was someone awhile back that was messing around with creating Rust bindings for Avalonia as an experiment.  I would suggest giving his full blog a read since it goes into a lot of detail about all the limitations of AOT.

https://ericsink.com/native_aot/overview.html

Otherwise you can also probably use that weird version of C++ that can directly use .NET types. Another way would be to host the entire CLR inside your app and then push and read everything from the CLR (much like how many apps use Lua for scripting).

https://en.wikipedia.org/wiki/C%2B%2B/CLI
https://learn.microsoft.com/en-us/dotnet/framework/unmanaged-api/hosting/clr-hosting-interfaces

A less hacky way would be to instead just break the GUI out into a different process so you have a proper C# app and then the backend can be written in what ever you want. Then communicate between them using IPC/TCP.
